### ISSUE TITLE
Flush before read

### ISSUE THREAD

**Issue by burrbull** [oli-obk/rust-si#20]
`read!` macro doesn't work properly with `print!`.

This works for me
```rust
macro_rules! readf {
	($($arg:tt)*) => {{
		use std::io::Write;
		std::io::stdout().flush().unwrap();
		read!($($arg)*)
	}};
}

...

print!("Xn = ");
let xn: f64 = readf!();
```
Output:
```
Xn = 0.5
```

I think `read!` should flush `stdout` before reading.

**Comment by oli-obk** [comment]
Makes sense. Since you already put in the work in figuring out how it works, do you want to open a PR with your change?

**Comment by burrbull** [comment]
I'm not sure which macro I should add this in. Please do it yourself. Your code is complicated for me.

**Comment by RadsammyT** [comment]
sorry for the necroposting but I am considering doing a PR that implements this albeit as a different macro.
```rust
#[macro_export]
macro_rules! readf {
	($($arg:tt)*) => {{
		use std::io::Write;
		std::io::stdout().flush().unwrap();
		read!($($arg)*)
	}};
}

#[macro_export]
macro_rules! try_readf {
	($($arg:tt)*) => {{
		use std::io::Write;
		std::io::stdout().flush().unwrap();
		try_read!($($arg)*)
	}};
}

```
I am essentially a beginner at rust (and also git, as I haven't had much experience on prepping PRs), so unfortunately I won't be able to implement this directly onto ```read!()```, if that is alright with you.
@oli-obk your thoughts?

**Comment by RadsammyT** [comment]
this issue can be closed now, PR #44 was merged and an update was published sooooooo
